spec/specification.md

```diff
@@ -0,0 +1,82 @@
+
+* Images are modeled as groups of ND Dense TileDB arrays with integer dimensions and uint8 attribute, i.e., one dense array per image level. 
+* Labels are also modeled as groups of ND Dense TileDB arrays with integer dimensions and uint8 attribute, i.e., one dense array per image level. 
+* The arbitrary key-value metadata will be modeled as **TileDB group metadata**.
+
+A potential two image Biomedical Imaging dataset, composed of one or more TileDB array groups, would look as follows:
+
+```
+.                               # Root folder, potentially in S3.
+    image_1.tiledb              # One image (id=1) converted to a TileDB Group.
+    |-- __group                 # TileDB group directory. 
+    |-- __meta                  # Group Metadata, contains all group metadata in a Key-Value manner. Metadata include all kinds of group metadata needed based on NGFF.     
+    |-- l_0.tdb                 # Layer 0 modeled as a 2D-5D dense TileDB Array with integer dimensions and uint8 attribute.
+    |   |-- __commits
+    |   |-- __fragments
+    |   |   `-- __1661541316115_1661541316115_bd1ea43c738344fa8ce2357091a2e558_14
+    |   `-- __schema
+    |-- l_1.tdb                 # Layer 1 modeled as a 2D-5D dense TileDB Array with integer dimensions and uint8 attribute.
+    |   |-- __commits
+    |   |-- __fragments
+    |   |   `-- __1661541316135_1661541316135_e1fadcee168f4dc585eaf39d4ea42c33_14
+    |   `-- __schema
+    `-- l_2.tdb                 # Layer 2 modeled as a 2D-5D dense TileDB Array with integer dimensions and uint8 attribute.
+        |-- __commits
+        |-- __fragments
+        |   `-- __1661541316146_1661541316146_bb390adff7c5474e8024b76b0478abc7_14
+        `-- __schema
+    |-- labels                      # Labels converted toa TileDB Group.
+        |-- __group                 # TileDB label group directory.
+        |-- __meta                  # LabelGroup Metadata.
+        |-- l_0.tdb                 # Layer 0 modeled as a 2D-5D dense TileDB Array with integer dimensions and uint8 attribute.
+        |   |-- __commits
+        |   |-- __fragments
+        |   |   `-- __1661541316115_1661541316115_bd1ea43c738344fa8ce2357091a2e558_14
+        |   `-- __schema
+        
+    image_2.tiledb              # One image (id=2) converted to a TileDB Group.
+    |-- __group                 # TileDB group directory
+    |-- __meta                  # Group Metadata
+    |-- l_0.tdb                 # Layer 0 modeled as a 2D-5D dense TileDB Array with integer dimensions and uint8 attribute.
+    |   |-- __commits
+    |   |-- __fragments
+    |   |   `-- __1661541316115_1661541316115_bd1ea43c738344fa8ce2357091a2e558_14
+    |   `-- __schema
+    |-- l_1.tdb                 # Layer 1 modeled as a 2D-5D dense TileDB Array with integer dimensions and uint8 attribute.
+    |   |-- __commits
+    |   |-- __fragments
+    |   |   `-- __1661541316135_1661541316135_e1fadcee168f4dc585eaf39d4ea42c33_14
+    |   `-- __schema
+```
+
+## Metadata
```

Comment:
We need a list of metadata keys, and specify whether they are required or optional. The most important one is the version number, which should be mandatory.